ethereum/test/test_beefy_light_client.js

```diff
@@ -0,0 +1,95 @@
+const BigNumber = web3.BigNumber;
+const {
+  deployBeefyLightClient, signatureSubstrateToEthereum,
+  createMerkleTree, mine, catchRevert
+} = require("./helpers");
+const fixture = require('./fixtures/beefy-fixture-data.json');
+
+require("chai")
+  .use(require("chai-as-promised"))
+  .use(require("chai-bignumber")(BigNumber))
+  .should();
+
+const { expect } = require("chai");
+
+describe("Beefy Light Client", function () {
+  let accounts;
+  let owner;
+  let userOne;
+  let userTwo;
+  let userThree;
+
+  before(async function () {
+    accounts = await web3.eth.getAccounts();
+    owner = accounts[0];
+    userOne = accounts[1];
+    userTwo = accounts[2];
+    userThree = accounts[3];
+
+    this.timeout(10 * 1000)
+
+    this.validatorAddressOne = "0xE04CC55ebEE1cBCE552f250e85c57B70B2E2625b";
+    this.validatorAddressTwo = "0x25451A4de12dcCc2D166922fA938E900fCc4ED24";
+    this.validatorsMerkleTree = createMerkleTree([this.validatorAddressOne, this.validatorAddressTwo]);
+    this.validatorsLeaf0 = this.validatorsMerkleTree.getHexLeaves()[0];
+    this.validatorsLeaf1 = this.validatorsMerkleTree.getHexLeaves()[1];
+    this.validator0PubKeyMerkleProof = this.validatorsMerkleTree.getHexProof(this.validatorsLeaf0);
+    this.validator1PubKeyMerkleProof = this.validatorsMerkleTree.getHexProof(this.validatorsLeaf1);
+
+    this.beefyLightClient = await deployBeefyLightClient(this.validatorsMerkleTree.getHexRoot(), this.validatorsMerkleTree.getLeaves().length);
+  });
+
+  it("runs new signature commitment and complete signature commitment correctly", async function () {
+    const initialBitfield = await this.beefyLightClient.createInitialBitfield([0, 1], 2);
+    expect(printBitfield(initialBitfield)).to.eq('11')
+
+    await this.beefyLightClient.newSignatureCommitment(
+      fixture.commitmentHash,
+      initialBitfield,
+      signatureSubstrateToEthereum(fixture.signature0),
+      0,
+      this.validatorAddressOne,
+      this.validator0PubKeyMerkleProof
+    ).should.be.fulfilled;
+
+    const lastId = (await this.beefyLightClient.currentId()).sub(new web3.utils.BN(1));
+
+    await mine(45);
+
+    const bitfield = await this.beefyLightClient.createRandomBitfield(lastId);
+    expect(printBitfield(bitfield)).to.eq('11')
```

Comment:
It would be great to actually test a subset of validators, instead of all. Either we create a new fixture with 3 validators, or we reduce `THRESHOLD_NUMERATOR`, but that would affect the production version. 

What do you think?

Reply:
added task to https://trello.com/c/2SlTcLns/83-smart-contracts-relay-chain-light-client-unhappy-paths-sufficient-unit-tests